numba/typing/npydecl.py

```diff
@@ -378,9 +378,18 @@ def generic(self, args, kws):
         pysig = None
         if kws:
             if self.method_name == 'sum':
-                def sum_stub(arr, axis):
-                    pass
-                pysig = utils.pysignature(sum_stub)
+                if 'axis' in kws and not 'dtype' in kws:
+                    def sum_stub(arr, axis):
+                        pass
+                    pysig = utils.pysignature(sum_stub)
+                if 'dtype' in kws and not 'axis' in kws:
```

Comment:
Same comment as above on the negation, also, this could be an `elif` clause.

Reply:
fixed